### ISSUE TITLE
Deprecated warning with new hashicorp/aws v5.0.1 provider

### ISSUE THREAD

**Issue by khosro-copper** [terraform-aws-modules/terraform-aws-eks#2635]
With latest hashicorp/aws v5.0.1 provider as well as latest terraform-aws-eks module, when I run terraform plan, I see this deprecated warning for aws addons:

   ```
Warning: Argument is deprecated

   with aws_eks_addon.this["coredns"],
   on main.tf line 388, in resource "aws_eks_addon" "this":
  388:   resolve_conflicts        = try(each.value.resolve_conflicts, "OVERWRITE")

 The "resolve_conflicts" attribute can't be set to "PRESERVE" on initial
 resource creation. Use "resolve_conflicts_on_create" and/or
 "resolve_conflicts_on_update" instead

 (and one more similar warning elsewhere)
```
If update main.tf with recommended suggestion, it should work as I test it locally.
Thanks!

**Comment by vstthomas** [comment]
Sorry, I'm still seeing this post-v5+.

```hcl
Warning: Argument is deprecated

  with module.eks.aws_eks_addon.this["kube-proxy"],
  on .terraform/modules/eks/main.tf line 399, in resource "aws_eks_addon" "this":
 399:   resolve_conflicts        = try(each.value.resolve_conflicts, "OVERWRITE")

The "resolve_conflicts" attribute can't be set to "PRESERVE" on initial resource creation. Use "resolve_conflicts_on_create" and/or "resolve_conflicts_on_update" instead

(and 2 more similar warnings elsewhere)
```

Applies to [L399](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/ec454c51cb0026254b68cafe7aa9d05f873af990/main.tf#L399) and [L427](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/ec454c51cb0026254b68cafe7aa9d05f873af990/main.tf#L427).

```sh
% tf version
Terraform v1.6.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v5.26.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.3.2
+ provider registry.terraform.io/hashicorp/kubernetes v2.23.0
+ provider registry.terraform.io/hashicorp/time v0.9.1
+ provider registry.terraform.io/hashicorp/tls v4.0.4
```

**Comment by bryantbiggs** [comment]
thats because it hasn't been resolved in this module yet - it will be resolved once v20.x is released

**Comment by vstthomas** [comment]
sorry, paying too much attention to the version in the subject "hashicorp/aws `v5.0.1` provider" 😬 

do we have a rough ETA on v20.x?

**Comment by bryantbiggs** [comment]
It will be this year (2023)

**Comment by adeelamin15** [comment]
https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2833

**Comment by bryantbiggs** [comment]
unfortunately v20 is blocked until https://github.com/hashicorp/terraform-provider-aws/pull/35037 is added since thats the main component of v20

**Comment by MoustafaAMahmoud** [comment]
It seems [EKS access entry and policy association changes #35037](https://github.com/hashicorp/terraform-provider-aws/pull/35037) has been merged. Do we have any updates about the release date?

**Comment by bryantbiggs** [comment]
It did, and I started adding it but there were some issues that have been patched and will go out on 5.34.0
- https://github.com/hashicorp/terraform-provider-aws/pull/35381
- https://github.com/hashicorp/terraform-provider-aws/pull/35391

**Comment by antonbabenko** [comment]
This issue has been resolved in version 20.0.0 :tada:
